### ISSUE TITLE
Nice.

### ISSUE THREAD

**Issue by arthurwolf** [noco-ai/spellbook-docker#6]
Really cool project.

I'm working on something similar (structurally at least), a manga-to-anime pipeline.
It involves a lot of different steps/models, similar to this project:
* Pre processing (alignment, upscaling, coloring).
* Separating pages into panels.
* Ordering the panels in the right reading order (took so much more effort than I thought...)
* Segmentation (using segment-anything)
* Extracting bubbles, the tails of bubbles/their vector, faces, bodies, backgrounds. Most of that necessitated training custom models.
* Assigning a character identity to each face/body. 
* Making a naive association between faces and bubbles.
* Reading the text of bubbles.
* I feed all that data to GPT4-V, and ask it to "read" each panel, telling it what happened in previous panels, what bubble is associated with what face, etc, asking it to "understand" what is happening in the panel, and to "deduce" some associations between the items, the tone of voice, etc. I tried "just" asking GPT4-V to read manga pages without all the steps above, and it was terrible at it. But with all the provided info (which causes easily 10k-token prompts, just for the text), it gets much better at it. It's sort of "pre-chewing" the work for him.
* That's where I am at now, the next step is going to be generating voice (what I'm working on now, bark/whisper/other models), sound effects, and then generating animation and special effects, and finally assembling all that into video.

I'll be looking closer into your project, in particular how it's organized, thanks a lot for sharing.
I'd be curious if you have any insights on how you'd do manga reading if you had to.

Cheers!

![masked](https://github.com/noco-ai/spellbook-docker/assets/108821/30277f35-f742-4c88-86ca-4491f4905f37)

![panel](https://github.com/noco-ai/spellbook-docker/assets/108821/f251976a-e47c-45c5-b24b-da1cc17e80fb)

[prompt.json](https://github.com/noco-ai/spellbook-docker/files/14490276/prompt.json)

[prompt.txt](https://github.com/noco-ai/spellbook-docker/files/14490277/prompt.txt)

[reading.json](https://github.com/noco-ai/spellbook-docker/files/14490278/reading.json)

[response.txt](https://github.com/noco-ai/spellbook-docker/files/14490279/response.txt)

[result.json](https://github.com/noco-ai/spellbook-docker/files/14490280/result.json)

![6253](https://github.com/noco-ai/spellbook-docker/assets/108821/756e6443-4f9a-4974-a54b-3c3f4884562c)

![6254](https://github.com/noco-ai/spellbook-docker/assets/108821/9a1bac9a-fb4f-4add-9a9a-e389fe158273)

**Comment by arthurwolf** [comment]
![page-3461-ids](https://github.com/noco-ai/spellbook-docker/assets/108821/af00ec72-11f0-4c38-a1ae-15313b243d06)
![page-3462-ids](https://github.com/noco-ai/spellbook-docker/assets/108821/7832e61e-536a-43b2-b5d1-b9e784fe85b1)
![page-3463-ids](https://github.com/noco-ai/spellbook-docker/assets/108821/34dbf890-8bee-4304-9884-226212d62e83)
![page-3464-ids](https://github.com/noco-ai/spellbook-docker/assets/108821/d54e8b8f-0f66-4d6a-9502-726283cdd0e4)
![page-3465-ids](https://github.com/noco-ai/spellbook-docker/assets/108821/eb1eeb39-e012-47a6-aee2-33d92e6bd2ed)

**Comment by arthurwolf** [comment]
https://github.com/noco-ai/spellbook-docker/assets/108821/194a7878-4f30-4744-a9a8-f69f4a4f9591

[result.json](https://github.com/noco-ai/spellbook-docker/files/14490286/result.json)
[reading.json](https://github.com/noco-ai/spellbook-docker/files/14490287/reading.json)
[response.txt](https://github.com/noco-ai/spellbook-docker/files/14490289/response.txt)
[prompt.txt](https://github.com/noco-ai/spellbook-docker/files/14490290/prompt.txt)
[prompt.json](https://github.com/noco-ai/spellbook-docker/files/14490291/prompt.json)
![panel](https://github.com/noco-ai/spellbook-docker/assets/108821/fe479ce9-04d5-4ea3-95f1-6fa9529a0335)
![masked](https://github.com/noco-ai/spellbook-docker/assets/108821/b994d9ca-a4e0-43bf-ac4a-b63bcd82ec86)

**Comment by arthurwolf** [comment]
[result.json](https://github.com/noco-ai/spellbook-docker/files/14490298/result.json)
[reading.json](https://github.com/noco-ai/spellbook-docker/files/14490299/reading.json)
[response.txt](https://github.com/noco-ai/spellbook-docker/files/14490300/response.txt)
[prompt.txt](https://github.com/noco-ai/spellbook-docker/files/14490301/prompt.txt)
[prompt.json](https://github.com/noco-ai/spellbook-docker/files/14490302/prompt.json)
![panel](https://github.com/noco-ai/spellbook-docker/assets/108821/15054438-d5fd-42d7-aa6e-dc6462b3da16)
![masked](https://github.com/noco-ai/spellbook-docker/assets/108821/7b1e836f-0025-4908-a854-a4133bda0281)

**Comment by noco-ai** [comment]
Hello! Your approach looks good to me, and it sounds like your hard work is paying off. If I was working on this particular project, I would experiment with fine tuning llava once you have a solid dataset to see if it gives better results than OpenAI's models. I have yet to see anyone share a finetune of llava for a specific task, so am curious how well it would work. If you are posting your progress on your project anywhere, please share the link as I am interest to see it in action once you have it all working.

**Comment by arthurwolf** [comment]
Thanks for the feedback.

I'll soon have about two comic books worth of data which I think would be enough to start fine tuning llava, but I have two issues there: 1. this is all very new, and there are no "easy guide" to fine tuning most things, llava even less, it's all very cryptic and assuming a high technical level, and 2. my assumption for llava is that even fine tuning would require a lot more compute than I can afford.

I've tried an alternative to this: trying to get my data into the llava training dataset for the next llava version. I've opened github issues and sent some emails but so far no answer. I hope I can make it happen, I think it'd benefit not just me, but the model itself also.

About posting progress ,I'm considering starting a youtube channel with some updates, I'll post about it here if/when that happens.

Cheers, and thanks again.
